### PR TITLE
feat: block automerge github-action-merge-dependabot major version

### LINKED ISSUE #110: Block automerge itself for semver major
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

Refs:
- https://github.com/fastify/github-action-merge-dependabot/pull/107#issuecomment-980005591
- #103 

When we release a major version of this GHA, dependabot will open a PR against this GHA users.
The `github-action-merge-dependabot` will merge the PR updating itself assuming users has the default `target: any` configuration.

This must not happen: a major bump of this GHA must be done manually by the user because a major release (could) mean a new GHA syntax.

The `github-action-merge-dependabot` cannot upgrade itself regardless the `target` filter.

In this case, we should:

- if the `api_url` is the default value:
  - do the normal flow: the external application will take care of rejecting the PR after landing https://github.com/fastify/dependabot-merge-action-app/issues/105
- if the `api_url` is not the default value:
  - set the GHA step as FAILED with a `how to upgrade` message
  - log the same message as a warning
  - (optional if possible) comment the PR with the same message

### PR DESCRIPTION
**Pull request by Eomm**
fixes #110 

Inspired by https://github.com/fastify/github-action-merge-dependabot/pull/107

### CODE REVIEW

**Comment by simoneb** [comment]
Can we please link to an issue?

File: test/action.test.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    inputs: { PR_NUMBER, TARGET: 'any', EXCLUDE_PKGS: ['foo'] }
+  })
+
+  await action()
+
+  t.ok(stubs.logStub.logInfo.calledOnceWith('foo is excluded, skipping.'))
+  t.ok(stubs.fetchStub.notCalled)
+})
+
+tap.test('apply only should not merge', async t => {
```
Comment (simoneb) [inline test/action.test.js:131]:
```suggestion
tap.test('approve only should not merge', async t => {
```

**Review by simoneb — COMMENTED** [review]
Do we have a test which checks that if you're not in the context of a PR but if you provide a [`pr-number`](https://github.com/fastify/github-action-merge-dependabot#pr-number) input the action works?

**Comment by Eomm** [comment]
> Do we have a test which checks that if you're not in the context of a PR but if you provide a pr-number input the action works?

Yeah, it was covered but not clear.
Added a new test case for this scenario

**Review by simoneb — APPROVED** [review]

**Pull request merged**